### PR TITLE
dev-scheme/racket: bump to 6.11, bug #636776

### PR DESCRIPTION
**Pull request by chrootsu**

### CODE REVIEW

**Comment by candrews** [comment]
Please update the commit message (see [GLEP 66](https://www.gentoo.org/glep/glep-0066.html#commit-messages)):
* `Closes` includes the meaning of `Bug` so both are unnecessary
* No need to include the bug number in the summary

File: dev-scheme/racket/racket-6.11.ebuild [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+		# digging through the Makefile in src/racket to find out which
+		# targets would build those binaries but not use them.
+		pushd racket
+		emake cgc-core
+		pax-mark m .libs/racketcgc
+		pushd gc2
+		emake all
+		popd
+		pax-mark m .libs/racket3m
+		popd
```
Comment (SoapGentoo) [inline dev-scheme/racket/racket-6.11.ebuild]: all these `pushd` and `popd` calls are missing `|| die`

File: dev-scheme/racket/racket-6.11.ebuild [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+}
+
+src_install() {
+	default
+
+	if use jit; then
+		# The final binaries need to be pax-marked, too, if you want to
+		# actually use them. The src_compile marking get lost somewhere
+		# in the install process.
+		for f in mred mzscheme racket; do
```
Comment (SoapGentoo) [inline dev-scheme/racket/racket-6.11.ebuild]: declare f local before

File: dev-scheme/racket/racket-6.11.ebuild [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+			pax-mark m "${D}/usr/bin/${f}"
```
Comment (SoapGentoo) [inline dev-scheme/racket/racket-6.11.ebuild]: `${D}` -> `${ED%/}`

File: dev-scheme/racket/racket-6.11.ebuild [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+		done
+
+		use X && pax-mark m "${D}/usr/$(get_libdir)/racket/gracket"
```
Comment (SoapGentoo) [inline dev-scheme/racket/racket-6.11.ebuild]: ditto

File: dev-scheme/racket/racket-6.11.ebuild (hunk 1 above)
Comment (SoapGentoo) [reply dev-scheme/racket/racket-6.11.ebuild]: also, in general its customary to redirect `pushd` and `popd` calls to `>/dev/null`

File: dev-scheme/racket/racket-6.11.ebuild [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+#
+#   * bug 426316
+#   * bug 486016
+#
+# The Racket GUI has some additional dependencies,
+#
+#   http://docs.racket-lang.org/gui/libs.html
+#
+# that may truly be runtime-only.
+#
```
Comment (Amynka) [inline dev-scheme/racket/racket-6.11.ebuild]: Please do not make essay from ebuilds

File: dev-scheme/racket/racket-6.11.ebuild [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+# dev-tex/slatex that has been modified to support only mzscheme. It's
+# not clear if dev-tex/slatex can be used as a replacement for the
+# racket version, but the racket version almost certainly cannot replace
+# dev-tex/slatex.
+#
+# If dev-tex/slatex will work for racket, then maybe we could just pull
+# it in as a dependency and remove /usr/bin/slatex in the src_install
+# for racket. Otherwise, we may have to rename racket's version to
+# somethine like slatex-racket, assuming that doesn't break
+# anything. This is all bug 547398.
```
Comment (Amynka) [inline dev-scheme/racket/racket-6.11.ebuild]: Please do not make essay from ebuilds.

File: dev-scheme/racket/racket-6.11.ebuild [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+
+# The main license is LGPL-3, as described here:
+#
+#   https://download.racket-lang.org/license.html
+#
+# However, there are traces of plain-GPL code, such as the routines in
+# collects/file/gzip.rkt that were based on GPLed C code, or the Cygwin
+# code in src/racket/dynsrc/init.cc. To err on the side of correctness,
+# we list GPL-3+, too.
+#
```
Comment (Amynka) [inline dev-scheme/racket/racket-6.11.ebuild]: This is not necessary at all.

**Pull request closed without merging**